src/sqlfluff/rules/ambiguous/AM08.py

```diff
@@ -0,0 +1,107 @@
+"""Implementation of Rule AM08."""
+
+from typing import Optional, Tuple
+
+from sqlfluff.core.parser import (
+    ComparisonOperatorSegment,
+    KeywordSegment,
+    LiteralSegment,
+    WhitespaceSegment,
+)
+from sqlfluff.core.rules import BaseRule, LintFix, LintResult, RuleContext
+from sqlfluff.core.rules.crawlers import SegmentSeekerCrawler
+
+
+class Rule_AM08(BaseRule):
+    """Join clauses should be present.
+
+    **Anti-pattern**
+
+    Cross joins are valid, but rare in the wild - and more often created by mistake than on purpose.
+    This rule catches situations where a cross join has been specified,
+    but not explicitly and so the risk of a mistaken cross join is highly likely.
+
+    .. code-block:: sql
+       :force:
+
+        SELECT
+            foo
+        FROM bar
+        JOIN baz;
+
+    **Best practice**
+
+    Use CROSS JOIN.
+
+    .. code-block:: sql
+       :force:
+
+        SELECT
+            foo
+        FROM bar
+        CROSS JOIN baz;
+    """
```

Comment:
```suggestion
    .. note::
       For SQL dialects which do not support an explicit :code:`CROSS JOIN`
       syntax we recommend using an inline ignore comments to mark a cross
       join as intentional:
       
       .. code-block:: sql

          SELECT
              foo
          FROM bar
          JOIN baz  -- noqa: ambiguous.join_condition    
    """
```

Reply:
@alanmcruickshank 

IMO we recommend comma syntax instead.
```sql
SELECT foo FROM bar, baz
```

Are you aware of any syntax dialects which does not support CROSS JOIN? I failed to find one in codebase.

Reply:
I'm not aware of one either, so it's a fairly academic discussion. It's probably ok to not add this note then 👍

Reply:
FWIW, I found a Stack Overflow post which says that `CROSS JOIN` is part of ANSI-92 SQL. _Hopefully_, that means everybody has it by now. 😅